### PR TITLE
Fixes #15822: IllegalArgumentException: Missing scheme when deleting user

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/15822

### CODE REVIEW

File: user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala [hunk 1]
```diff
@@ -142,15 +143,19 @@ object UserManagementService {
...
-        val srcXML = ConstructingParser.fromFile(File(path).toJava, preserveWS = true).document.children
-        val digest =  new PasswordEncoder.DigestEncoder((srcXML \\ "authentication" \ "@hash").text)
-        (srcXML \\ "authentication").head match {
-          case e: Elem =>
-            val newXml = e.copy(child = e.child ++ newUser.copy(password = getHash((srcXML \\ "authentication" \ "@hash").text).encode(newUser.password)).toNode)
-            UserManagementIO.replaceXml(srcXML, newXml, path)
-            Full(newUser)
-          case _ =>
-            Failure(s"Wrong formatting : $path")
+        tryo(ConstructingParser.fromFile(File(path).toJava, preserveWS = true)) match {
```
Comment (fanf) [inline user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala]: you can avoid pattern matching here with a `map`:
```
tryo(...).map { parsedFile => 
... 
}
```

File: user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala [hunk 2]
```diff
@@ -160,15 +165,19 @@ object UserManagementService {
...
-        val srcXML = ConstructingParser.fromFile(File(path).toJava, preserveWS = true).document.children
-        val toUpdate = (srcXML \\ "authentication").head
-        val newXml = new RuleTransformer(new RewriteRule {
-          override def transform(n: Node): NodeSeq = n match {
-            case user: Elem if (user \ "@name").text == toDelete => NodeSeq.Empty
-            case other => other
-          }
-        }).transform(toUpdate).head
-        UserManagementIO.replaceXml(srcXML, newXml, path)
+       tryo(ConstructingParser.fromFile(File(path).toJava, preserveWS = true)) match {
```
Comment (fanf) [inline user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala]: Same as above

File: user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala [hunk 3]
```diff
@@ -66,11 +67,11 @@ object UserManagementIO {
...
   def getHash(s: String) = {
     s.toLowerCase match {
-      case "sha" | "sha1"       => PasswordEncoder.SHA1
+      case "sha" | "sha1" => PasswordEncoder.SHA1
       case "sha256" | "sha-256" => PasswordEncoder.SHA256
       case "sha512" | "sha-512" => PasswordEncoder.SHA512
-      case "md5"                => PasswordEncoder.MD5
-      case _                    => PasswordEncoder.PlainText
+      case "md5" => PasswordEncoder.MD5
+      case _ => PasswordEncoder.PlainText
```
Comment (fanf) [inline user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala]: why breaking the formating ?

**Comment by ElaadF** [comment]
OK, merging this PR

**Pull request merged**